src/server/serverWrappers.cpp

```diff
@@ -0,0 +1,475 @@
+    }
+    close(sock);
+}
+
+/**
+ * Method called when a UDP read notification was received.
+ * It reads from the socket and sends it off for processing.
+ * John Agapeyev March 19
+ */
+void readUDP(const int sock, sockaddr *servaddr, socklen_t *servAddrLen) {
+    int nmesg;
+    if ((nmesg = recvmmsg(sock, udpMesgs, MAX_UDP_PACKET_COUNT, MSG_DONTWAIT, nullptr)) == -1) {
+        perror("recvmmsg");
+        return;
+    }
+
+    logv("Received %d messages\n", nmesg);
+
+#pragma omp parallel for schedule(static) firstprivate(nmesg)
+    for (int i = 0; i < nmesg; ++i) {
+        processPacket(readBuffers[i]);
+    }
+}
+
+/**
+ * Waits on an epoll descriptor for events to occur.
+ * It writes the event data to the epoll event buffere that was passed in.
+ * It then returns the number of events generated.
+ * John Agapeyev March 19
+ */
+int waitForEpollEvent(const int epollfd, epoll_event *events) {
+    int nevents;
+    if ((nevents = epoll_wait(epollfd, events, MAXEVENTS, -1)) == -1) {
+        perror("epoll_wait");
+        exit(1);
+    }
+    return nevents;
+}
+
+/**
+ * Creates the epoll event buffer for use in the epoll read loops.
+ * John Agapeyev March 19
+ */
+epoll_event *createEpollEventList() {
+    epoll_event *events;
+    if (!(events = reinterpret_cast<epoll_event *>(calloc(MAXEVENTS, sizeof(epoll_event))))) {
+        perror("Calloc failure");
+        exit(1);
+    }
+    return events;
+}
+
+/**
+ * Processes a received username.
+ * Due to the nature of TCP, a new client will be conencted, and then send its username seperately.
+ * This method registers the username and updates its entry in the client list.
+ * After reading the username, it sends the client back its username with its generated id.
+ * Once the client has been updated, it then messages all other clients, informing them of a newly connected player.
+ * yPos is made static to ensure all clients start at different positions instead of being on top of one another
+ * John Agapeyev March 19
+ */
+void processClientUsername(const int sock, const char *buff, std::pair<const int32_t, PlayerJoin>& client) {
+    static float yPos = 0;
+    //Handle initial username read
+    client.second.hasSentUsername = true;
+    client.second.isPlayerReady = false;
+    strncpy(client.second.entry.username, buff + TCP_HEADER_SIZE + 1, NAMELEN);
+    strcat(client.second.entry.username, "\0");
+    logv("Server received username: %s\n", client.second.entry.username);
+
+    const size_t bufferSize = NAMELEN + TCP_HEADER_SIZE + 1;
+    char outBuff[bufferSize];
+    memset(outBuff, '\0', bufferSize);
+    int32_t *id = reinterpret_cast<int32_t *>(outBuff);
+    *id = client.first;
+    outBuff[4] = 'C';
+    outBuff[5] = '/';
+
+    strncpy(outBuff + TCP_HEADER_SIZE + 1, client.second.entry.username, NAMELEN);
+
+    gm->createMarine(client.first);
+    auto& marine = gm->getMarine(client.first).first;
+    marine.setPosition(100, yPos);
+    yPos += marine.getW() * 10;
```

Comment:
now it might make more sense but it's still a lot. really it should be pulled from the map